File: Python/10.soliddiamond.py
'''Create this pattern

    * 
   * * 
  * * * 
 * * * * 
* * * * * 
* * * * * 
 * * * * 
  * * * 
   * * 
    * 

'''

N = int(input("Enter the n. of rows: "))

for row in range(N):
    for col in range(N-row-1):
        print(" ", end='')
        
    for col in range(row+1):
        print("* ", end='')
        
    print("\n")
    
    
for row in range(N):
    for col in range(row):
        print(" ", end='')
        
    for col in range(N-row):
        print("* ", end='')
        
    print("\n")
    

File: C++/10.soliddiamond.cpp
/*Create this pattern

    * 
   * * 
  * * * 
 * * * * 
* * * * * 
* * * * * 
 * * * * 
  * * * 
   * * 
    * 
    
*/


#include <iostream>

using namespace std;

int main()
{
    int N;
    cout<<"Enter No. of Rows:";
    cin>>N;
    
    for (int row = 0; row<N; row++){
        for (int col =0; col< (N-row-1); col++){
            cout<<" ";
        }
        
        for (int col = 0; col < row +1; col++){
            cout<<"* ";
        }
        cout<<endl;
        
    }
    
    for(int row=0; row<N; row++){
        for (int col =0; col<row; col++){
            cout<<" ";
        }
        
        for(int col = 0; col<N-row; col++){
            cout<<"* ";
        }
        cout<<endl;
    }
    
    

    return 0;
}